common/test/unit/com/thoughtworks/go/domain/PipelineConfigTest.java

```diff
@@ -236,15 +239,71 @@ public void shouldGetAllTemplateVariableNames() {
     }
 
     @Test
+    public void shouldValidateCorrectPipelineLabelWithoutAnyMaterial() {
+        PipelineConfig pipelineConfig = new PipelineConfig(new CaseInsensitiveString("cruise"), new MaterialConfigs());
+        pipelineConfig.setLabelTemplate("pipeline-${COUNT}-alpha");
+        pipelineConfig.validate(null);
+        assertThat(pipelineConfig.errors().isEmpty(), is(true));
+        assertThat(pipelineConfig.errors().on(PipelineConfig.LABEL_TEMPLATE), is(nullValue()));
+    }
+
+    private PipelineConfig createAndValidatePipelineLabel(String labelFormat) {
+        GitMaterialConfig git = new GitMaterialConfig("git@github.com:gocd/gocd.git");
+        git.setName(new CaseInsensitiveString("git"));
+
+        PipelineConfig pipelineConfig = new PipelineConfig(new CaseInsensitiveString("cruise"), new MaterialConfigs(git));
+        pipelineConfig.setLabelTemplate(labelFormat);
+
+        pipelineConfig.validate(null);
+
+        return pipelineConfig;
+    }
+
+    @Test
+    public void shouldValidateCorrectPipelineLabelWithoutTruncationSyntax() {
+        String labelFormat = "pipeline-${COUNT}-${git}-454";
+        PipelineConfig pipelineConfig = createAndValidatePipelineLabel(labelFormat);
+        assertThat(pipelineConfig.errors().on(PipelineConfig.LABEL_TEMPLATE), is(nullValue()));
+    }
+
+    @Test
+    public void shouldValidatePipelineLabelWithNonExistingMaterial() {
+        String labelFormat = "pipeline-${COUNT}-${NoSuchMaterial}";
+        PipelineConfig pipelineConfig = createAndValidatePipelineLabel(labelFormat);
+        assertThat(pipelineConfig.errors().on(PipelineConfig.LABEL_TEMPLATE), startsWith("You have defined a label template in pipeline"));
+    }
+
+    @Test
+    public void shouldValidateCorrectPipelineLabelWithTruncationSyntax() {
+        String labelFormat = "pipeline-${COUNT}-${git[:7]}-alpha";
```

Comment:
maybe it is a good idea to add the unit tests anyway -- of course due to the XSD check there cannot be such a configuration.